eclass/gkrellm-plugin-r1.eclass

```diff
@@ -0,0 +1,108 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: gkrellm-plugin-r1.eclass
+# @MAINTAINER:
+# maintainer-needed@gentoo.org
+# @AUTHOR:
+# Original author: Jim Ramsay
+#      -r1 author: David Seifert
+# @BLURB: Provides common methods used by (almost) all gkrellm plugins
+# @DESCRIPTION:
+# - Sets up default dependencies
+# - Provides utility routines in lieu of hard-coding the plugin directories.
+# - Provides the most common src_install method to avoid code duplication.
+#
+# Changelog:
+#   01 January 2018: David Seifert <soap@gentoo.org>
+#     - Port to EAPI 6, remove built_with_use
+#   12 March 2007: Jim Ramsay <lack@gentoo.org>
+#     - Added server plugin support
+#   09 March 2007: Jim Ramsay <lack@gentoo.org>
+#     - Initial commit
+#
+
+# @ECLASS-VARIABLE: PLUGIN_SO
+# @DESCRIPTION:
+# The name of the plugin's .so file which will be installed in
+# the plugin dir. Defaults to "${PN}.so". Has to be a bash array.
+
+# @ECLASS-VARIABLE: PLUGIN_SERVER_SO
+# @DESCRIPTION:
+# The name of the plugin's server plugin .so portion. Unset by default.
+# Has to be a bash array.
+
+# @ECLASS-VARIABLE: PLUGIN_DOCS
+# @DESCRIPTION:
+# An optional list of docs to be installed. Has to be a bash array.
+
+case ${EAPI:-0} in
+	[0-5])
+		die "${ECLASS} is banned in EAPI ${EAPI:-0}"
+		;;
+	6)
+		;;
+	*)
+		die "Unknown EAPI ${EAPI:-0}"
+		;;
+esac
+
+EXPORT_FUNCTIONS src_install
+
+if [[ ! ${_GKRELLM_PLUGIN_R1} ]]; then
+
+DEPEND="virtual/pkgconfig"
+
+# @FUNCTION: _gkrellm-plugin-r1_dir
+# @INTERNAL
+# @DESCRIPTION:
+# Return the path of the plugin directory.
+_gkrellm-plugin-r1_dir() {
+	echo /usr/$(get_libdir)/gkrellm2/plugins
+}
+
+# @FUNCTION: _gkrellm-plugin-r1_server_dir
+# @INTERNAL
+# @DESCRIPTION:
+# Return the path of the server plugin directory.
+_gkrellm-plugin-r1_server_dir() {
+	echo /usr/$(get_libdir)/gkrellm2/plugins-gkrellmd
+}
+
+# @FUNCTION: gkrellm-plugin-r1_src_install
+# @DESCRIPTION:
+# Install the plugins and call einstalldocs
+gkrellm-plugin-r1_src_install() {
+	insinto $(_gkrellm-plugin-r1_dir)
+
+	if ! declare -p PLUGIN_SO >/dev/null 2>&1 ; then
+		doins ${PN}.so
```

Comment:
`doexe` is more correct for `.so` files. Also, don't you want to use `get_modname` here?